src/variables/components/VariableDropdownReactSelect.tsx

```diff
@@ -0,0 +1,369 @@
+     * if the string is set to 'open', and then the user closes it, and the code sets it to open again,
+     * unless the code sets it to something else in between (like null or 'close'),
+     * then nothing will happen- the menu will not open)
+     */
+
+    if (menuOpen !== prevMenuOpen && menuOpen !== null) {
+      newState['menuOpen'] = null
+    }
+
+    // need to have this, as the 'onClickAwayHere' gets triggered *before*
+    // the selected value is set to the actualValue (it keeps re-using the original
+    // property)
+
+    // for updating the selected value:
+    if (justSelected()) {
+      newState = {...newState, typedValue: actualVal, selectHappened: false}
+    }
+
+    this.setState(newState)
+  }
+
+  getInitialValuesAfterLoading() {
+    const {values} = this.props
+    // it reloaded; maybe because of a dependent var
+    // want to re-init
+
+    return {
+      shownValues: values,
+      loaded: true,
+    }
+  }
+
+  render() {
+    const {selectedValue, name, status} = this.props
+    const {shownValues} = this.state
+
+    const isDisabled = !shownValues || shownValues.length === 0
+
+    let realVals = []
+
+    //  make them all look like: { value: 'vanilla', label: 'Vanilla' }
+    if (shownValues && shownValues.length) {
+      realVals = shownValues.map(val => ({value: val, label: val}))
+    }
+    const placeHolderText = this.getPlaceHolderText('Select a Value')
+
+    const foreground = 'white'
+    const inputBackground = 'black'
+    const background = '#2c2d35'
+    const focusColor = '#46454e'
+    const selectColor = '#ce58eb'
+
+    const customStyles = {
+      option: (provided, state) => {
+        let backgroundColor = state.isFocused ? focusColor : background
+
+        if (state.isSelected) {
+          backgroundColor = selectColor
+        }
+
+        return {
+          ...provided,
+          color: foreground,
+          backgroundColor,
+          padding: 10,
+        }
+      },
+      clearIndicator: provided => {
+        // console.log("clear indicator...state??? ACK", state)
+        // const foregroundColor = state.isFocused? 'yellow' : 'cyan'
+        return {
+          ...provided,
+          '&:hover': {
+            color: 'white',
+          },
+        }
+      },
+      control: () => ({
+        width: 225,
+        height: 40,
+        color: foreground,
+        padding: 3,
+        backgroundColor: background,
+        display: 'flex',
+      }),
+      valueContainer: provided => ({
+        ...provided,
+        color: foreground,
+        backgroundColor: inputBackground,
+      }),
+      placeHolder: provided => ({
+        ...provided,
+        color: foreground,
+        backgroundColor: inputBackground,
+      }),
+      indicatorSeparator: () => ({
+        display: 'none',
+      }),
+      dropdownIndicator: (provided, state) => {
+        const hoverBackground = state.isFocused ? '#5d5f6f' : '#828497'
```

Comment:
can you pull these two hex values out into variables at the top of the file along with the other colors please?